pom.xml

```diff
@@ -36,8 +36,8 @@
 	<properties>
 		<project.parent.version>${project.version}</project.parent.version>
 		<!-- Dependency versions -->
-		<spring-cloud-dependencies.version>2023.0.3</spring-cloud-dependencies.version>
-		<spring-boot-dependencies.version>3.3.4</spring-boot-dependencies.version>
+		<spring-cloud-dependencies.version>2024.0.0</spring-cloud-dependencies.version>
+		<spring-boot-dependencies.version>3.4.1</spring-boot-dependencies.version>
```

Comment:
Let's please also make adjustments to documentation as necessary for this release.

* Root README table
* Documentation 'Getting Started' table: https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/docs/src/main/asciidoc/getting-started.adoc
* If any of these changes are significant enough to warrant it, consider a migration doc similar to https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/main/docs/src/main/asciidoc/migration-guide-3.x.adoc